tests/integration/api_v3/http/posStakes.test.js

```diff
@@ -31,47 +31,40 @@ const endpoint = `${baseUrlV3}/pos/stakes`;
 
 describe('Stakes API', () => {
 	let refValidator;
+	let maxNumberSentStakes;
 	beforeAll(async () => {
-		let response;
+		const posConstants = await api.get(`${baseUrlV3}/pos/constants`);
+		maxNumberSentStakes = posConstants.data.maxNumberSentStakes;
+
 		do {
 			// eslint-disable-next-line no-await-in-loop
-			response = await api.get(`${baseUrlV3}/dpos/delegates?limit=1`);
-		} while (!response.data);
-		[refValidator] = response.data;
+			const { data: [stakeTx] = [] } = await api.get(`${baseUrlV3}/transactions?moduleCommand=pos:stake&limit=1`);
+			if (stakeTx) {
+				refValidator = stakeTx.sender;
+			}
+		} while (!refValidator);
```

Comment:
Why do we need a while loop here? This could cause the test to halt till timeout.